rts/motoko-rts/src/memory/ic/linear_memory.rs

```diff
@@ -39,7 +42,7 @@ impl Memory for IcMemory {
         let delta = u64::from(bytes.as_u32());
 
         // Update heap pointer
-        let old_hp = u64::from(HP);
+        let old_hp = u64::from(getHP());
```

Comment:
It would be interesting to see how many cycles could be saved if we migrate this function back to the `compile.ml` side.

Reply:
I think a better way is to inline the `getHP` and `setHP` functions at link time into the code directly.

Reply:
I'll revert my experiment shortly.